### PR TITLE
Fs helper for temporary files and directories

### PR DESCRIPTION
**Pull request by cclerget**
**Description of the Pull Request (PR):**

- moved `MakeTmpDir` previously in e2e internal package as `fs` helper package and also added `MakeTmpFile`.
- moved cache test function in a separate package `test/tool` to avoid import cycle issue and updated package relying on those functions.
- modified `SetCacheDir` to use `MakeTmpDir`.

**This fixes or addresses the following GitHub issues:**

- Fixes partially #3770 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by gvallee — APPROVED** [review]
LGTM

File: internal/pkg/test/tool/cache.go [hunk 1]
```diff
@@ -3,13 +3,13 @@
 // LICENSE.md file distributed with the sources of this project regarding your
 // rights to use or distribute this software.
 
-package test
+package tool
```
Comment (mem) [inline internal/pkg/test/tool/cache.go]: Please add documentation for this package. In particular it's necessary to differentiate between `github.com/sylabs/singularity/internal/pkg/test` and `github.com/sylabs/singularity/internal/pkg/test/tool` from the point of view of someone adding code to either of those packages. Think of it like this: the name "test" implies that that package contains everything related to tests, while the name "tool" is generic, so some guidance is required as to what belongs in `tool` and not in `test`.

**Review by mem — CHANGES_REQUESTED** [review]

File: internal/pkg/test/tool/cache.go (hunk 1 above)
Comment (cclerget) [reply internal/pkg/test/tool/cache.go]: @mem added in https://github.com/sylabs/singularity/blob/243d57aaa91ff430b980e5f3d9ec086bcc7086eb/internal/pkg/test/tool/doc.go

File: e2e/internal/e2e/fileutil.go [hunk 2]
```diff
@@ -10,6 +10,8 @@ import (
 	"io/ioutil"
 	"os"
 
+	"github.com/sylabs/singularity/internal/pkg/util/fs"
+
```
Comment (WestleyK) [inline e2e/internal/e2e/fileutil.go]: Why is there a extra newline here?

**Review by WestleyK — COMMENTED** [review]
Looks good, just one comment about a newline :+1:

File: e2e/internal/e2e/fileutil.go (hunk 2 above)
Comment (cclerget) [reply e2e/internal/e2e/fileutil.go]: a vscode tool too lazy as me ;). Fixed

File: internal/pkg/util/fs/helper.go [hunk 3]
```diff
@@ -6,6 +6,8 @@
 package fs
```
Comment (WestleyK) [inline internal/pkg/util/fs/helper.go:7]: Copyright should be `2018-2019` in this file.

File: internal/pkg/util/fs/helper_linux_test.go [hunk 4]
```diff
@@ -9,6 +9,7 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
```
Comment (WestleyK) [inline internal/pkg/util/fs/helper_linux_test.go:14]: Same here, `2018-2019`

**Review by WestleyK — COMMENTED** [review]
Just a couple suggestions about the copyright year _(not strictly necessary)_, otherwise looks good to me :+1:

**Review by WestleyK — APPROVED** [review]
Looks good to me :+1:

**Review by mem — APPROVED** [review]
LGTM

**Pull request merged**